lib/utils.js

```diff
@@ -427,13 +427,38 @@ function normalizeLicenseExpression(rawLicenseExpression, logger) {
   // therefore use the mapped license expression as an argument if it was found
   const mappedLicenseExpression = scancodeMap.get(rawLicenseExpression)
   const parsed = SPDX.parse(mappedLicenseExpression || rawLicenseExpression || '', licenseVisitor)
-  const result = SPDX.stringify(parsed)
 
+  // normalize the parsed license expression will recursively normalize the parsed license expression
+  const normalizedParsed = _normalizeParsedLicenseExpression(parsed, logger)
+
+  const result = SPDX.stringify(normalizedParsed)
   if (result === 'NOASSERTION') logger.info(`ScanCode NOASSERTION from ${rawLicenseExpression}`)
 
   return result
 }
 
+function _normalizeParsedLicenseExpression(parsedLicenseExpression, logger) {
```

Comment:
Would this logic be more appropriate in [SPDX](https://github.com/clearlydefined/spdx/blob/096c8069217450b676434688da0f106a5182675d/index.js#L29C12-L29C31), or in in [spdx-expression-parse](https://github.com/clearlydefined/spdx-expression-parse.js/blob/fork/parse.js)? Similar to licenseVisitor in [parseLicense](https://github.com/clearlydefined/spdx-expression-parse.js/blob/4d69bb5b95e540da61a4c5c625e204eaedb4eff7/parse.js#L88), does introducing a licenseRefVisitor at parseLicenseRef help?

Reply:
Draft PR see https://github.com/clearlydefined/spdx-expression-parse.js/pull/9